### DISCUSSION TITLE
Tasmota do not succeed to connect to Mosquitto broker

### DISCUSSION THREAD

**Discussion by jackpotcityco** [arendst/Tasmota#19727] [Q&A] [answered]
Hello,

**Program Version	13.1.0.4(tasmota)
Build Date & Time	2023-10-09T14:35:26
Core/SDK Version	2_7_4_9/2.2.2-dev(38a443e)**

I am using:
**ESP8266EX
VL53L1X - Time-of-Flight (ToF) ranging sensor**

I am trying to use MQTT from Tasmota Web Interface but the problem is that it seems that it never succeeds to connect to the 
MQTT broker (Mosquitto) and can't understand what is causing this. I will explain in steps what I have done below.

1. Below is the error that happens all the time:
   Initial timeout was 200 ms. I have changed that to 5000 ms in the console to give connection more time: `MqttWifiTimeout 5000`
![image](https://github.com/arendst/Tasmota/assets/47510843/f389dc8a-6590-4bf2-89fa-18220dffd416)

**Step 2 to 5 below indicates that the MQTT "**Mosquitto**" broker is working properly:**

2. I have "**Mosquitto**" running (RED Rectangle)
3. I have managed to connect to the "**Mosquitto**" using a Python script on topic: `client.subscribe("tasmota_01")`
    _I connect using: 127.0.0.1 and port 1883 without username and password (GREEN Rectangle)_
4. I have managed to connect to the "**Mosquitto**" using MQTT-Explorer
    _I connect using: 127.0.0.1 and port 1883 without username and password (BLUE Rectangle)_
5. I have managed from MQTT-Explorer Publish: "**tasmota_01/Hello from MQTT Explorer**" (PURPLE Rectangle)
![image](https://github.com/arendst/Tasmota/assets/47510843/af7e9789-4113-40cb-8c3a-2db982532119)

6. I have "MQTT enable" checked as seen below in the Web Interface.
![image](https://github.com/arendst/Tasmota/assets/47510843/210f07c1-1c50-4b82-ac67-35bb80ba2f89)

7. * I connect using: 127.0.0.1 and port 1883 as seen below in the Web Interface.. 
    * I do not use any username or password even that I have put in the fields below.
    * The "mosquitto.conf" has the original settings, which do not use: username and password.
![image](https://github.com/arendst/Tasmota/assets/47510843/d1c6cd00-3ec3-4bd5-9fab-bf1970e9fa68)

8. I do not have any VPN, Antivirus or Firewall enabled.

I am not sure if it is possible to understand why Tasmota do not connect to "**Mosquitto**" ?

Thank you!

**Comment by barbudor** [comment, marked as answer]
127.0.0.1 is the local host address
When used on your PC it means 'the PC'
When used on Tasmota, it means 'that device'

So as your mosquitto is running on a PC, you need to configure tasmota to connect to the actual ip address of your PC, not 127.0.0.1

Also beware that mosquitto V2 do not listen by default to external connection.
Read that article: http://www.steves-internet-guide.com/mosquitto-broker/

**Reply by jackpotcityco** [reply]
That makes sense, then I understand. I managed to be able to connect with Tasmota to the Broker now. 

Below are the steps I did to make it work:

In ipconfig I could see my PCs internal address as:
`15.1.0.17`

I then put that like this and saved configuration:
![image](https://github.com/arendst/Tasmota/assets/47510843/7a6fd08b-ef4c-4b80-bbf7-2e39f7f2a066)

In the `mosquitto.conf` file under # Listeners section I added the below 2 lines, to allow `mosquitto `to allow for any external connections on port 1883. 
```
listener 1883 0.0.0.0
allow_anonymous true
```

I opened up port 1883 in the Firewall to allow for connctions on that port.

I then started the broker:
![image](https://github.com/arendst/Tasmota/assets/47510843/0be8a294-9d1c-4610-ad16-f53d2f3e4cb7)

Now the web interface could connect:
![image](https://github.com/arendst/Tasmota/assets/47510843/c2745f4b-4a0b-4963-a550-e259371de71a)

Thank you for your help!
